node/core/dispute-coordinator/src/real/ordering/mod.rs

```diff
@@ -279,7 +279,20 @@ impl OrderingProvider {
 				rx.await.or(Err(Fatal::ChainApiSenderDropped))?.map_err(Fatal::ChainApi)?
 			};
 
-			let earliest_block_number = head_number - hashes.len() as u32;
+			let earliest_block_number = match head_number.checked_sub(hashes.len() as u32) {
+				Some(number) => number,
+				None => {
+					// It's assumed that it's impossible to retrieve
+					// more than N ancestors for block number N.
+					tracing::error!(
+						target: LOG_TARGET,
+						"Received {} ancestors for block number {} from Chain API",
+						hashes.len(),
+						head_number,
+					);
+					return Ok(ancestors)
+				},
```

Comment:
It is a bit weird to return `Ok` for an error in a function returning `Result`.

Reply:
This branch is considered to be unreachable, so we log this at `error` level and return valid ancestors that we managed to fetch.
It does feel weird but still better than introducing an error variant for the match arm that can only happen in case of a bug in chain API

Reply:
We actually have a quite a lot of those (errors that should actually be unreachable), but you have a point.